src/Cli/PhpCommands.php

```diff
@@ -36,47 +23,60 @@ public function whoami($options = ['as' => 'default'])
      */
     public function phpCookbookUpdate($options = ['as' => 'default'])
     {
-        $token = $this->getGitHubToken($options['as']);
+        $api = $this->api($options['as']);
 
         $rpmbuild_php_url = $this->getConfig()->get('projects.rpmbuild-php.repo');
         $rpmbuild_php_dir = $this->getConfig()->get('projects.rpmbuild-php.path');
 
         $php_cookbook_url = $this->getConfig()->get('projects.php-cookbook.repo');
         $php_cookbook_dir = $this->getConfig()->get('projects.php-cookbook.path');
 
-        $rpmbuild_php = new WorkingCopy($rpmbuild_php_url, $rpmbuild_php_dir);
+        $rpmbuild_php = WorkingCopy::clone($rpmbuild_php_url, $rpmbuild_php_dir, $api);
         $rpmbuild_php
             ->setLogger($this->logger);
 
         // Look at the most recent commit on the current branch.
-        $output = $rpmbuild_php->show('HEAD');
+        $output = $rpmbuild_php->git('show HEAD');
 
         // Look for changes to both the php_version and rpm_datecode in the same file.
         $version_updates = [];
         $version = false;
         foreach ($output as $line) {
             if (preg_match('#^diff #', $line)) {
                 $version = false;
-            }
-            elseif (preg_match('#\+%define php_version *([0-9.]*)#', $line, $matches)) {
+            } elseif (preg_match('#.%define php_version *([0-9.]*)#', $line, $matches)) {
                 $version = $matches[1];
-            }
-            elseif (preg_match('#\+%define rpm_datecode *([0-9]*)#', $line, $matches)) {
+            } elseif ($version && preg_match('#\+%define rpm_datecode *([0-9]*)#', $line, $matches)) {
                 $rpm_datecode = $matches[1];
                 $version_updates[$version] = "{$version}-{$rpm_datecode}";
             }
         }
 
         // If there were no updates to php versions, then we are done.
         if (empty($version_updates)) {
-            $this->say("Nothing was updated.");
+            $this->logger->notice("Nothing was updated.");
             return;
         }
 
-        $php_cookbook = new WorkingCopy($php_cookbook_url, $php_cookbook_dir);
+        $php_cookbook = WorkingCopy::clone($php_cookbook_url, $php_cookbook_dir, $api);
         $php_cookbook
             ->setLogger($this->logger);
 
+        // Create a commit message with all of our modified versions
+        $all_updated_versions = $this->prettyImplode(', ', ' and ', array_map(function ($v) {
+            return "php-$v";
+        }, array_keys($version_updates)));
+        $preamble = $this->preamble();
+        $message = "{$preamble}{$all_updated_versions}";
+        $this->logger->notice("Commit message {message}", ['message' => $message]);
+
+        // Create a set of version ids from the commit message
+        $vids = new VersionIdentifiers();
```

Comment:
Here is where we define the pattern that will determine whether an existing pull request is an exact or partial match for the pull request we plan to open.